subworkflows/local/scaffold_curation/main.nf

```diff
@@ -0,0 +1,331 @@
+        ch_hifi,
+        getPrimaryAssembly( ch_assemblies ),
+        keySet: ['id','sample'],
+        meta: 'rhs'
+    )
+    .multiMap { meta, hifi_reads, fasta -> 
+        hifi_reads: [ meta, hifi_reads ]
+        reference : fasta 
+    }
+    .set{ asm_hifi_ch }
+
+    joinByMetaKeys( 
+        merge_hifi_bam.singleton
+            .map { meta, bam -> [ meta, *bam ] } // the spread operator (*) flattens the bam list
+            .mix( SAMTOOLS_MERGE_HIFI.out.bam),
+        CREATE_CHROMOSOME_SIZES_FILE.out.sizes,
+        keySet: ['id','sample'],
+        meta: 'rhs'
+    )
+    .multiMap { meta, bam, chrom_sizes -> 
+        merged_bam   : [ meta, bam ] 
+        chrom_sizes  : chrom_sizes
+    }
+    .set { bam2coverage_ch }
+
+    // Coverage-track: create hitile- and bed-coverage tracks 
+    BAM2COVERAGE_TRACKS(
+        bam2coverage_ch.merged_bam,
+        bam2coverage_ch.chrom_sizes
+    )
+    ch_versions  = ch_versions.mix( BAM2COVERAGE_TRACKS.out.versions )
+
+    // Gap-track: create bedtrack 
+    SEQTK_CUTN(getPrimaryAssembly( ch_assemblies ))
+    ch_versions  = ch_versions.mix( SEQTK_CUTN.out.versions )
+
+    // Gap-track: create beddb- and bed-gap tracks 
+    joinByMetaKeys( 
+        SEQTK_CUTN.out.bed,
+        CREATE_CHROMOSOME_SIZES_FILE.out.sizes,
+        keySet: ['id','sample'],
+        meta: 'rhs'
+    )
+    .multiMap { meta, bed, chrom_sizes -> 
+        bed          : [ meta, bed ] 
+        chrom_sizes  : chrom_sizes
+    }
+    .set { bed2gap_ch }
+
+    CREATE_GAP_TRACKS(
+        bed2gap_ch.bed,
+        bed2gap_ch.chrom_sizes
+    ) 
+    ch_versions  = ch_versions.mix( CREATE_GAP_TRACKS.out.versions )
+
+    // Telomer-track: create track
+    TIDK_SEARCH_BEDGRAPH(
+        getPrimaryAssembly( ch_assemblies ),
+        params.telomer_motif
+    )
+    ch_versions  = ch_versions.mix( TIDK_SEARCH_BEDGRAPH.out.versions )
+
+    TIDK_SEARCH_TSV(
+        getPrimaryAssembly( ch_assemblies ),
+        params.telomer_motif
+    )
+    ch_versions  = ch_versions.mix( TIDK_SEARCH_TSV.out.versions )
+
+    // Telomer-track: create plot - thats not really necessary, but nice to have 
+    TIDK_PLOT(TIDK_SEARCH_TSV.out.tsv)
+    ch_versions  = ch_versions.mix( TIDK_PLOT.out.versions )
+
+    // Telomer-track: convert telomer bedgraph into beddb file that can be ingested into HiGlass 
+    joinByMetaKeys( 
+        TIDK_SEARCH_BEDGRAPH.out.bedgraph,
+        CREATE_CHROMOSOME_SIZES_FILE.out.sizes,
+        keySet: ['id','sample'],
+        meta: 'rhs'
+    )
+    .multiMap { meta, bedgraph, chrom_sizes -> 
+        bedgraph          : [ meta, bedgraph ] 
+        chrom_sizes  : chrom_sizes
+    }
+    .set { bedgraph_telomer_ch }
+
+    CREATE_TELOMER_HITILE_TRACK(
+        bedgraph_telomer_ch.bedgraph,
+        bedgraph_telomer_ch.chrom_sizes
+    )
+    ch_versions  = ch_versions.mix( CREATE_TELOMER_HITILE_TRACK.out.versions )
+
+    // ingest coverage, gap and telomer track into Pretext
+    joinByMetaKeys(
```

Comment:
Maybe add a comment here generally by these functions on what's expected in the cross. Are we doing 1-1, 1-*, or *-*.

Reply:
For each assembly, there will always be one pretext map, one coverage track, one telomere track and one gap track. So in principle it should be always 1-1. 

At some point we need to think about how to handle diploid/polyploid assemblies. We can run this as is and create the tracks separately on each haploid assembly. But we can also combine multiple haplo-assemblies into a new diploid/polyploid assembly and create the coverage tracks for this new assembly. Also in that case it should be 1-1-1-1.